recipes/qarchive/all/conanfile.py

```diff
@@ -0,0 +1,156 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+from textwrap import dedent
+
+class qarchiveConan(ConanFile):
+    name = "qarchive"
+    license = "BSD-3-Clause"
+    homepage = "https://antonyjr.in/QArchive/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("QArchive is a cross-platform C++ library that modernizes libarchive ,\
+                    This library helps you to extract and compress archives supported by libarchive")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, "fPIC": True}
+    generators = "cmake", "cmake_find_package"
+    topics = ("conan", "qt", "Qt", "compress", "libarchive")
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            if self.settings.os == "Windows" and self.options.shared:
+                self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
You can explicitly _always_ set this in the cmake_wrapper since it handles all the logic about applying it